src/main/java/com/vmware/vipclient/i18n/messages/service/ProductService.java

```diff
@@ -77,13 +73,13 @@ public ProductService(BaseDTO dto) {
      *
      * @return list of locales of the product specified in the dto object
      */
-    public List<String> getSupportedLocales(){
-        List<String> locales = null;
+    public Set<String> getSupportedLocales(){
+        Set<String> locales = new HashSet<>();
         Iterator<DataSourceEnum> msgSourceQueueIter = VIPCfg.getInstance().getMsgOriginsQueue().iterator();
-        while((locales == null || locales.isEmpty()) && msgSourceQueueIter.hasNext()){
+        while(msgSourceQueueIter.hasNext()){
```

Comment:
After remove the null or empty judgement, it will loop all the data source no matter get the locale list or not in the former data source.

Reply:
It seems the returned locales are a join of local and remote.

Reply:
Yes. If offline mode is supported too, then it means that locales that are found only in offline are supported as well and must be loaded.

Reply:
Will there be such case 'locales found only in offline'?